src/psyclone/tests/domain/lfric/transformations/dynamo0p3_transformations_test.py

```diff
@@ -2879,45 +2877,47 @@ def test_multi_builtins_red_then_fuse_do(tmpdir, monkeypatch, annexed,
                     in result)
             code = (
-                "      asum = 0.0_r_def\n"
-                "      !\n"
-                "      !$omp parallel default(shared), private(df)\n"
-                "      !$omp do schedule(static), reduction(+:asum)\n"
-                "      DO df = loop0_start, loop0_stop, 1\n"
-                "        ! Built-in: X_innerproduct_Y (real-valued fields)\n"
-                "        asum = asum + f1_data(df) * f2_data(df)\n"
-                "        ! Built-in: inc_a_times_X (scale a real-valued field)"
+                "    asum = 0.0\n"
+                "\n"
+                "    ! Call kernels and communication routines\n"
+                "    !$omp parallel default(shared), private(df)\n"
+                "    !$omp do schedule(static), reduction(+:asum)\n"
+                "    do df = loop0_start, loop0_stop, 1\n"
+                "      ! Built-in: X_innerproduct_Y (real-valued fields)\n"
+                "      asum = asum + f1_data(df) * f2_data(df)\n"
                 "\n"
-                "        f1_data(df) = bsum * f1_data(df)\n"
-                "      END DO\n"
-                "      !$omp end do\n"
-                "      !$omp end parallel\n"
-                "      !\n"
-                "      ! Set halos dirty/clean for fields modified in the "
+                "      ! Built-in: inc_a_times_X (scale a real-valued field)"
+                "\n"
+                "      f1_data(df) = bsum * f1_data(df)\n"
+                "    enddo\n"
+                "    !$omp end do\n"
+                "    !$omp end parallel\n"
+                "\n"
+                "    ! Set halos dirty/clean for fields modified in the "
                 "above loop(s)\n"
-                "      !\n"
-                "      CALL f1_proxy%set_dirty()\n"
-                "      !\n"
-                "      ! End of set dirty/clean section for above loop(s)\n"
-                "      !\n"
-                "      global_sum%value = asum\n"
-                "      asum = global_sum%get_sum()\n")
+                "    call f1_proxy%set_dirty()\n"
+                "\n"
+                "    ! Perform global sum\n"
+                "    global_sum%value = asum\n"
+                "    asum = global_sum%get_sum()\n")
         else:  # not distmem, annexed is True or False
             assert "loop0_stop = undf_aspc1_f1" in result
             code = (
-                "      asum = 0.0_r_def\n"
-                "      !\n"
-                "      !$omp parallel default(shared), private(df)\n"
-                "      !$omp do schedule(static), reduction(+:asum)\n"
-                "      DO df = loop0_start, loop0_stop, 1\n"
-                "        ! Built-in: X_innerproduct_Y (real-valued fields)\n"
-                "        asum = asum + f1_data(df) * f2_data(df)\n"
-                "        ! Built-in: inc_a_times_X (scale a real-valued field)"
+                "    asum = 0.0\n"
```

Comment:
Strictly, this should still be `0.0_r_def`. If this is easy to fix then please add the precision, otherwise it could be delayed to another PR (with a TODO).

Reply:
Done (it was easier than I thought)